src/NServiceBus.Core/Routing/UnicastPublishRouter.cs

```diff
@@ -69,38 +59,22 @@ static IEnumerable<UnicastRoutingTarget> SelectDestinationsForEachEndpoint(IDist
             }
         }
 
-        async Task<IEnumerable<IUnicastRoute>> GetDestinations(ContextBag contextBag, Type[] typesToRoute)
+        async Task<IEnumerable<UnicastRoutingTarget>> GetDestinations(ContextBag contextBag, Type[] typesToRoute)
         {
             var messageTypes = typesToRoute.Select(t => new MessageType(t));
             var subscribers = await subscriptionStorage.GetSubscriberAddressesForMessage(messageTypes, contextBag).ConfigureAwait(false);
-            return subscribers.Select(s => new SubscriberDestination(s));
+            return subscribers.Select(s => CreateRoutingTarget(s));
```

Comment:
I wanted to do further changes to that class because there are quite a few more LINQ usages. Planned to raise a new PR for that though.